Password encryption/marwan

batainem96 opened the pull request:
Implemented password encryption using Salt/Base64 algorithms via JavaTPoint implementation. Incorporated encryption into login (findUserByCredentials: AppUser) and register (save: AppUser) UserRepository functions. Included some legacy logic to authenticate users still using old DB schema (no encryption) in the aforementioned login function.

batainem96 committed:
Added Salt and Base64 encryption based class for password encryption from javatpoint (link in class doc for PasswordUtils)

batainem96 committed:
Cleaned up imports in AppUser

batainem96 committed:
Updated code style conventions and comments

batainem96 committed:
Updated citation to be clearer.

wsingleton commented:
this is a good start, but lets do a few things before we merge this in:

- store a common key used for encryption in a properties file rather than storing it alongside the password within the datasource. 
- remove the logic that checks for "legacy" users without an encrypted password (it's just dummy data, we can update it)
- remove breadcrumb sysouts from console output

batainem96 committed:
Removed legacy handling, sout statements, and moved salt value to application.properties as a common key.

batainem96 commented:
You should be able to use any string for the salt property, really, but if not --- just call PasswordUtils.getSalt(30) for an equivalent key.

I've cleaned up compatibility code, sout statements, and incorporated a common key in the properties file for encryption/validation. Register and Login both work for me with encrypted passwords.

wsingleton committed:
Merge branch 'main' into password_encryption/marwan

wsingleton commented:
merged, good work @batainem96
